src/pages/CarDetails.jsx

```diff
@@ -1,9 +1,72 @@
-import React from 'react';
+import { useParams, Link } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import axios from 'axios';
 
-const CarDetails = () => (
-  <div>
-    <h1>Car Details page will be displayed here</h1>
-  </div>
-);
+const CarDetailPage = () => {
+  const [car, setCar] = useState({});
+  const { carId } = useParams();
 
-export default CarDetails;
+  useEffect(() => {
+    const getSingleCar = async () => {
+      try {
+        const res = await axios.get(
+          `http://127.0.0.1:3001/api/v1/cars/${carId}`,
+        );
+        setCar(res.data);
+      } catch (error) {
+        console.error('Error fetching car:', error);
+      }
+    };
+    getSingleCar();
+  }, [carId]);
```

Comment:
Please add the `authorization token` in the headers of the fetch request so that the proper data is returned from the API.
```suggestion
    const getSingleCar = async () => {
      try {
        const headers = { authorization: getToken() };
        const res = await axios.get(
          `http://127.0.0.1:3001/api/v1/cars/${carId}`, { headers },
        );
        setCar(res.data);
      } catch (error) {
        console.error('Error fetching car:', error);
      }
    };
    getSingleCar();
  }, [carId]);
```